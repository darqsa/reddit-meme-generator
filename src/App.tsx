import { Navigate, Route, Routes } from "react-router-dom";
import Footer from "./components/Footer/Footer";
import Header from "./components/Header/Header";
import HomePage from "./pages/HomePage";

function App() {
  return (
    <>
      <Header />
      <Routes>
        <Route path="/" element={<Navigate to="/home" />} />
        <Route path="/home" element={<HomePage />} />
      </Routes>
      <Footer />
    </>
  );
}

export default App;
