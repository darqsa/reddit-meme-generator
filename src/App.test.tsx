import { MemoryRouter } from "react-router-dom";
import App from "./App";
import TestRenderer from "react-test-renderer";

describe("Given an App component", () => {
  describe("When it's instantiated in home", () => {
    test("Then it show Home page", () => {
      const expectedApp = TestRenderer.create(
        <MemoryRouter initialEntries={["/home"]}>
          <App />
        </MemoryRouter>
      );

      expect(expectedApp).toMatchSnapshot();
    });
  });

  describe("When it's instantiated in a unkown page", () => {
    test("Then it should show the notFoundPage", () => {
      const expectedApp = TestRenderer.create(
        <MemoryRouter initialEntries={["*"]}>
          <App />
        </MemoryRouter>
      );

      expect(expectedApp).toMatchSnapshot();
    });
  });

  describe("When it's instantiated in favorites", () => {
    test("Then it should show the favorites page", () => {
      const expectedApp = TestRenderer.create(
        <MemoryRouter initialEntries={["/favorites"]}>
          <App />
        </MemoryRouter>
      );

      expect(expectedApp).toMatchSnapshot();
    });
  });
});
