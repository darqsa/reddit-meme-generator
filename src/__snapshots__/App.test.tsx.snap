// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given an App component When it's instantiated in a unkown page Then it should show the notFoundPage 1`] = `
Array [
  <div
    className="sc-crXcEl kMesAo header-container"
  >
    <h1
      className="sc-evZas grHgol header-container__title"
    >
      Page Not Found
    </h1>
    <a
      href="/home"
      onClick={[Function]}
    >
      <button
        className="sc-dkzDqf wbqHG"
      >
        Return to Main Page
      </button>
    </a>
  </div>,
  <div
    className="sc-bjUoiL fkMwtq modal-container"
  >
    <div
      className="modal-container__popup--error"
    >
      <img
        alt="Sad meme frog"
        className="modal-container__image"
        src="img/errorModal.png"
      />
      <span>
        Something went wrong, I'm sad...
      </span>
    </div>
  </div>,
  <div
    className="sc-bczRLJ MqMTL footer-container"
  >
    <span
      className="footer__text"
    >
      <span
        className="footer__text--especial"
      >
        ©
      </span>
      Tamarindos curiosos. inc
      <img
        alt="imagen de la ranita"
        className="footer__image"
        src="favicon.png"
      />
    </span>
  </div>,
]
`;

exports[`Given an App component When it's instantiated in favorites Then it should show the favorites page 1`] = `
Array [
  <div
    className="header-container"
  >
    <header
      className="sc-hKMtZM cklVyW header-container__title-container"
    >
      <h1
        className="sc-jSMfEi kZvA-DQ header-container__title"
      >
        FAVORITE 
        <span
          className="sc-papXJ eRKGXE"
        >
          MEMES
        </span>
      </h1>
    </header>
    <section
      className="sc-gKXOVf iqGkus header-container__section-container"
    >
      <div
        className="sc-iBkjds eITUJN header-container__texts-container"
      >
        <p
          className="sc-ftvSup jccDpC header-container__text"
        >
          Here you can find your selected favorite memes! Click delete to remove a meme from your list.
        </p>
      </div>
      <div
        className="sc-jqUVSM jOaidU header-container__buttons-container"
      >
        <a
          href="/home"
          onClick={[Function]}
        >
          <button
            className="sc-dkzDqf wbqHG"
          >
            Main Page
          </button>
        </a>
      </div>
    </section>
  </div>,
  <div
    className="sc-ksZaOG diyRTk"
  >
    <ul
      className="meme-list"
    />
  </div>,
  <div
    className="sc-bczRLJ MqMTL footer-container"
  >
    <span
      className="footer__text"
    >
      <span
        className="footer__text--especial"
      >
        ©
      </span>
      Tamarindos curiosos. inc
      <img
        alt="imagen de la ranita"
        className="footer__image"
        src="favicon.png"
      />
    </span>
  </div>,
]
`;

exports[`Given an App component When it's instantiated in home Then it show Home page 1`] = `
Array [
  <div
    className="header-container"
  >
    <header
      className="sc-hKMtZM cklVyW header-container__title-container"
    >
      <img
        alt="Awesome meme generator logo, it's a frog"
        className="sc-eCYdqJ dFUjSg header-container__title-image"
        src="img/logo.png"
      />
      <h1
        className="sc-jSMfEi kZvA-DQ header-container__title"
      >
        AWESOME 
        <span
          className="sc-papXJ eRKGXE"
        >
          MEME 
        </span>
        GENERATOR
      </h1>
    </header>
    <section
      className="sc-gKXOVf iqGkus header-container__section-container"
    >
      <div
        className="sc-iBkjds eITUJN header-container__texts-container"
      >
        <p
          className="sc-ftvSup jccDpC header-container__text"
        >
          Click the “Generate memes” button to get 10 random memes, if you like a meme you can click the
           
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-star header-container__text-icon"
            data-icon="star"
            data-prefix="fas"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 576 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M381.2 150.3L524.9 171.5C536.8 173.2 546.8 181.6 550.6 193.1C554.4 204.7 551.3 217.3 542.7 225.9L438.5 328.1L463.1 474.7C465.1 486.7 460.2 498.9 450.2 506C440.3 513.1 427.2 514 416.5 508.3L288.1 439.8L159.8 508.3C149 514 135.9 513.1 126 506C116.1 498.9 111.1 486.7 113.2 474.7L137.8 328.1L33.58 225.9C24.97 217.3 21.91 204.7 25.69 193.1C29.46 181.6 39.43 173.2 51.42 171.5L195 150.3L259.4 17.97C264.7 6.954 275.9-.0391 288.1-.0391C300.4-.0391 311.6 6.954 316.9 17.97L381.2 150.3z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
           
          button to move it into your favourite pages.
        </p>
        <p
          className="sc-ftvSup jccDpC header-container__text"
        >
          In your favourite pages you can unlike a meme, edit it and also create your own meme.
        </p>
      </div>
      <div
        className="sc-jqUVSM jOaidU header-container__buttons-container"
      >
        <button
          className="sc-dkzDqf wbqHG"
          onClick={[Function]}
        >
          Generate Memes
        </button>
        <a
          href="/favorites"
          onClick={[Function]}
        >
          <button
            className="sc-dkzDqf wbqHG"
          >
            My Favorites
          </button>
        </a>
      </div>
    </section>
  </div>,
  <div
    className="sc-ksZaOG diyRTk"
  >
    <ul
      className="meme-list"
    />
  </div>,
  <div
    className="sc-bczRLJ MqMTL footer-container"
  >
    <span
      className="footer__text"
    >
      <span
        className="footer__text--especial"
      >
        ©
      </span>
      Tamarindos curiosos. inc
      <img
        alt="imagen de la ranita"
        className="footer__image"
        src="favicon.png"
      />
    </span>
  </div>,
]
`;
